# Lab 5 - compare with genomes from other species

START HERE

<!--Maybe we should add genomes from species which are actually quite different (on top of the four) (e.g. Saccharomyces cerevisiae as that would be one more model organism). Then fill a table (which genes in which species). Then do alignments and possibly compare different genes e.g. cytb versus another one? (versus a nuclear gene?)
Is it the right moment to compare to NUMTs? See this paper: Mitochondrial DNA-like sequences in the nucleus (NUMTs): insights into our African origins and the mechanism of foreign DNA integration [there is a table listing the genes in which they found NUMT based on similarity with mtDNA. And coordinates for where the NUMT are in these genes. But not sure which reference genome they used...]. Some NUMT are present in more than one copy. Different ages of NUMTS. They could each choose a NUMT from the table of the paper (if we know how to convert to the right reference genome that is...). Actually blast seem to be strategy to identify NUMT (see also Bensasson et al 2001) so fits well with the whole tutorial. But there are other approaches, e.g. in this 2019 paper: A Novel Phylogenetic Approach for De Novo Discovery of Putative Nuclear Mitochondrial (pNumt) Haplotypes [the total of putative NUMT does not change dramatically though - a few hundreds]. Or Sanger sequencing (coverage, double peaks etc).
There is also a list here: https://www.mitomap.org/foswiki/bin/view/MITOMAP/PseudogeneList (a bit more recent) (and https://www.mitomap.org/foswiki/bin/view/MITOMAP/WebHome has a lot of resources but will be down this Friday). If we want to make a connection to diseases it could be here (see wikipedia page for a first overview).

Which alignment tool should we use? So many questions... I'm going to stop for today.

Comment: discovery that organelles have separate DNA: in the sixties.-->

<!--In the worst case use the alignment tutorial from last year (for pairwise alignment).-->

## Introduction / Background information to lab 4

## Goals

## Input(s)

## Output(s)

## Tools

## Steps

## Details