# Session 5 - compare with genomes from other species

## Introduction / Background information to Session 5

In this session you will focus on aligments. As you have seen in the lecture, it is possible to align two sequences (pairwise alignments) and multiple sequences (multiple alignments). You will do a bit of both. For the pairwise alignment part, you will follow a tutorial that was developped by Rasmus Wernersson. For the multiple aligments, you will go back to working with the mitochondrial genomes from sessions 3 and 4. You will also work with additional mitochondrial genomes. This will prepare you for the bioinformatic project, as one of the first steps of the project will be to align sequences!

## Goals

  + Perform local and global pairwise alignments with different algorithms
  + Explore parameters of pairwise alignments
  + Perform multiple alignments with different softwares

## Input(s)

  + Mitochondrial genomes from the following species: XX

## Output(s)

  + An alignment of XX mitochondrial genome
  + A table of mitochondrial features for X genomes
  + Some kind of visual representation of the alignment

## Tools

  + Online alignment tools : https://www.ebi.ac.uk/Tools/psa/
  + Online tool to randomly shuffle a protein sequence: http://www.bioinformatics.org/sms2/shuffle_protein.html

## Steps

  + Start by doing the pairwise alignment exercise.
  + 

## Details

### Pairwise alignment

Please go through the tutorial on this page: http://teaching.bioinformatics.dtu.dk/teaching/index.php/Exercise:_Pairwise_alignment

Questions are numbered from 1 to 14. For the report, you need to submit written answers to the following questions: 9, 11, 14, **more?**.
